dynatrace/api/v1/config/customtags/list/list_test.go

```diff
@@ -0,0 +1,158 @@
+package list
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"sync/atomic"
+	"testing"
+	"time"
+
+	"github.com/dynatrace-oss/terraform-provider-dynatrace/dynatrace/rest"
+)
+
+const (
+	NUM_ENTITY_TYPES = 50
+	NUM_CUSTOM_TAGS  = 50
+	NUM_ENTITIES     = 50
+)
+
+type TestClient struct {
+	latency     time.Duration // artificial per-request delay inside Finish
+	inFlight    int64         // current in-flight
+	maxInFlight int64         // max observed in-flight
+	total       int64         // total Finish calls
+}
+
+func NewTestClient(latency time.Duration) *TestClient {
+	return &TestClient{latency: latency}
+}
+
+func (client *TestClient) Get(ctx context.Context, url string, expectedStatusCodes ...int) rest.Request {
+	return &Request{client: client, ctx: ctx}
+}
+func (client *TestClient) Post(ctx context.Context, url string, payload any, expectedStatusCodes ...int) rest.Request {
+	panic("unsupported operation")
+}
+func (client *TestClient) Put(ctx context.Context, url string, payload any, expectedStatusCodes ...int) rest.Request {
+	panic("unsupported operation")
+}
+func (client *TestClient) Delete(ctx context.Context, url string, expectedStatusCodes ...int) rest.Request {
+	panic("unsupported operation")
+}
+func (client *TestClient) Credentials() *rest.Credentials {
+	panic("unsupported operation")
+}
+
+// Accessors for assertions
+func (c *TestClient) MaxConcurrent() int64 { return atomic.LoadInt64(&c.maxInFlight) }
+func (c *TestClient) Total() int64         { return atomic.LoadInt64(&c.total) }
+func (c *TestClient) InFlight() int64      { return atomic.LoadInt64(&c.inFlight) }
+
+type Request struct {
+	client *TestClient
+	ctx    context.Context
+}
+
+func (request *Request) Finish(v ...any) error {
+	c := request.client
+
+	// increment in-flight
+	cur := atomic.AddInt64(&c.inFlight, 1)
+
+	// update maxInFlight if needed (lock-free CAS loop)
+	for {
+		max := atomic.LoadInt64(&c.maxInFlight)
+		if cur <= max {
+			break
+		}
+		if atomic.CompareAndSwapInt64(&c.maxInFlight, max, cur) {
+			break
+		}
+	}
```

Comment:
Is the loop really needed or would this be sufficient?
```suggestion
	max := atomic.LoadInt64(&c.maxInFlight)
	if cur > max {
		atomic.CompareAndSwapInt64(&c.maxInFlight, max, cur)
	}
```